### ISSUE TITLE
Enable overwriting of library names and paths for downstream developers

### ISSUE THREAD

**Issue by hmaarrfk** [cgohlke/imagecodecs#1]
Placeholder issue. and cross reference to discussion started 

https://github.com/conda-forge/staged-recipes/pull/10331#issuecomment-635608248

**Comment by cgohlke** [comment]
The plan for the next version is to allow distributors to modify the default build options/settings by providing a `distributor_setup.customize_build` function, which is imported and run in `setup.py`.
It would look like this for conda-forge (not tested):
```Python
# distributor_setup.py

import sys
import os


def customize_build(EXTENSIONS, OPTIONS):
    """Modify imagecodecs build options/settings for conda-forge."""

    library_inc = os.environ.get('LIBRARY_INC', '')

    del EXTENSIONS['jpeg12']
    del EXTENSIONS['jpegxl']
    del EXTENSIONS['zfp']

    # build jpeg8 against libjpeg instead of libjpeg_turbo
    OPTIONS['cythonize'] = True
    EXTENSIONS['jpeg8']['cython_compile_env']['HAVE_LIBJPEG_TURBO'] = False

    EXTENSIONS['jpegxr']['libraries'] = ['libjpegxr', 'libjxrglue']
    EXTENSIONS['jpegxr']['include_dirs'] = [
        os.path.join(library_inc, 'jpegxr')
    ]

    if sys.platform == 'win32':
        EXTENSIONS['bz2']['libraries'] = ['bzip2']
        EXTENSIONS['jpeg2k']['include_dirs'] = [
            os.path.join(
                library_inc, 'openjpeg-' + os.environ.get('openjpeg', '2.3')
            )
        ]
        EXTENSIONS['jpegls']['libraries'] = ['charls-2-x64']
        EXTENSIONS['lz4']['libraries'] = ['liblz4']
        EXTENSIONS['lzma']['libraries'] = ['liblzma']
        EXTENSIONS['png']['libraries'] = ['libpng', 'z']
        EXTENSIONS['webp']['libraries'] = ['libwebp']

```

This function could also be included in setup.py and executed conditionally (`if os.environ.get('LIBRARY_INC', '')`) if that is easier to maintain.

**Comment by cgohlke** [comment]
Should be fixed in v2020.5.30. 

https://github.com/cgohlke/imagecodecs/blob/f38d0d4e5db1d08d92742a5d899325fc7c7b56e7/setup.py#L268-L312

If the (untested) `customize_build_cf` function does not work, you can provide your own `imagecodecs_distributor_setup.customize_build` function.

**Comment by hmaarrfk** [comment]
Thanks!
